### DISCUSSION TITLE
Using Bluetooth speakers

### DISCUSSION THREAD

**Discussion by ProHarshit** [pschatzmann/arduino-espeak-ng#14] [Q&A] [answered]
Hi. I want to use the espeak project with Bluetooth. Reading the Audio Tools project, I think it is possible. But can anyone suggest me the proper code for that? I am using ESP32 WROOM 32

**Comment by pschatzmann** [comment, marked as answer]
You can e.g. use an [A2DPStream](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_a2_d_p_stream.html) as output class

**Comment by ProHarshit** [comment]
Can you please suggest the lines I have to change?

**Comment by pschatzmann** [comment]
Have a look at [this example](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-tts/streams-simple_tts-a2dp) and just change the [TTS implementation](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-tts/streams-espeak-i2s/streams-espeak-i2s.ino)

**Comment by ProHarshit** [comment]
Sir, will this code work? I am not able to run my computer for some temp issues. 
#include "AudioTools.h" 
#include "FileSystems.h"
#include "espeak.h"
#include "AudioLibs/A2DPStream.h"

const char* name = "LEXON MINO L"; 

AudioInfo from(22050, 1, 16);                                 
AudioInfo to(44100, 2, 16);  

A2DPStream a2dp; 
FormatConverterStream out(a2dp);
ESpeak espeak(a2dp);


void setup() {
Serial.begin(115200);
espeak.begin();
audio_info espeak_info = espeak.audioInfo();
Serial.println("Starting...");
out.begin(from, to);
auto cfg = a2dp.defaultConfig(TX_MODE); 
cfg.name = name;
cfg.silence_on_nodata = true;
a2dp.begin(cfg);
a2dp.setVolume(0.3);
Serial.println("A2DP Started");

void loop() {
  espeak.say("Hello world!");
  delay(5000);
}

**Comment by pschatzmann** [comment]
No, you need to send the output from the espeak to out and not a2dp!

**Comment by ProHarshit** [comment]
#include "AudioTools.h"
#include "FileSystems.h"
#include "espeak.h"
#include "AudioLibs/A2DPStream.h"

const char* name = "LEXON MINO L";

AudioInfo from(22050, 1, 16);
AudioInfo to(44100, 2, 16);

A2DPStream a2dp;
FormatConverterStream out(a2dp);
ESpeak espeak(out);

void setup() {
Serial.begin(115200);
espeak.begin();
audio_info espeak_info = espeak.audioInfo();
Serial.println("Starting...");
out.begin(from, to);
auto cfg = a2dp.defaultConfig(TX_MODE);
cfg.name = name;
cfg.silence_on_nodata = true;
a2dp.begin(cfg);
a2dp.setVolume(0.3);
Serial.println("A2DP Started");

void loop() {
espeak.say("Hello world!");
delay(5000);
}

**Reply by ProHarshit** [reply]
Is this ok sir?
